### PR TITLE
Add YAML support + Replacement in plain-text files

### PR DESCRIPTION
**Pull request by rkcreation**
Adds a special path (replace) for text/plain files, for example to replace badges for last tag number in README files.

Adds text/yaml support for yaml files (same logic than json files, but with js-yaml parser).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,13 +40,50 @@ The `out` option can also be an array of files:
...
-    "out": { "file": "manifest.json", "path": "deeper.current" }
+    "in": { "file": "VERSION", "type": "text/plain" },
+    "out": { "file": "VERSION", "type": "text/plain" }
   }
 }
 ```
+
+### Replacements
+
+Using the special path `replace` for `text/plain` files, the last version number is replaced by the new one.
```
Comment (webpro) [inline README.md]: I'm not sure about a better alternative, but using "replace" for `path` doesn't feel exactly right (in theory it even conflicts with actual intention of using this prop literally). Maybe something like a new attribute "mode" (maybe "global"?). But really, better suggestions are most welcome.

File: index.js [hunk 2]
```diff
@@ -53,7 +54,18 @@ class Bumper extends Plugin {
...
           return writeFile(file, JSON.stringify(parsed, null, indent) + '\n');
+        } else if (type === 'text/yaml') {
+          const data = await readFile(file, 'utf8').catch(() => '{}');
+          const indent = detectIndent(data).indent || '  ';
+          const parsed = yaml.safeLoad(data);
+          set(parsed, path, version);
+          return writeFile(file, yaml.safeDump(parsed, { indent: indent.length }) + '\n');
         } else if (type === 'text/plain') {
+          if ( path === 'replace' ) {
+            const data = await readFile(file, 'utf8').catch(() => '{}');
```
[the remark this answers is not in the document]
Comment (rkcreation) [reply index.js]: Right !

File: index.js [hunk 3]
```diff
@@ -53,7 +54,18 @@ class Bumper extends Plugin {
...
+            this.log.info(`Replacing ${this.config.contextOptions.latestVersion} by ${version} in ${file}...`)
+            return writeFile(file, data.replace(new RegExp(this.config.contextOptions.latestVersion.replace(/[.*+?^${}()|[\]\\]/g, "\\$&"), 'g'), version));
```
[the remark this answers is not in the document]
Comment (rkcreation) [reply index.js]: It's for escaping latestVersion and use it as regex pattern

File: README.md (hunk 1 above)
Comment (rkcreation) [reply README.md]: You're right, for cleaner and clearer options, it's better to use a separate option.  
As it's a feature which is on/off (replace entire file content or parts), I suggest something like mode (parts | full) or regex (true | false).

**Comment by webpro** [comment]
Thanks @rkcreation! I've merged your PR, and released this as v.1.1.0.

I think I've simplified things a bit by simply replacing the version string in the text file, whether it's a full file replacement or a greedy regex.

**Comment by rkcreation** [comment]
Great ! Your rewrite seems cleaner than my first attempt 😃 
Thanks for your work too @webpro !

**Pull request merged**